File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.py
import sys
from collections import Counter

def solution(nums):
    answer = 0
    cnt = len(nums)//2
    adict = Counter(nums)
    # print(adict)
    if len(adict) >= cnt :
        answer = cnt
    else :
        answer = len(adict)
    
    return answer

File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    var answer = 0;
    let map = new Map();
    for(let i of nums) {
        if(!map.has(i)) map.set(i, 0);
        if(map.has(i)) map.set(i, map.get(i) + 1);
    }
    if(map.size < Math.floor(nums.length / 2))
        return map.size;
    else return Math.floor(nums.length / 2)
        
}